### ISSUE TITLE
 rules in setting weights in the combination of conv2d layer and batch norm layer

### ISSUE THREAD

**Issue by tearsland123** [Lasagne/Lasagne#914]
Dear Sir:
  Really sorry to bother you, I could not fully understand the rules in setting weights in the combination of conv2d layer and batch norm layer, For example,
   net['conv'] = batch_norm(Conv2DLayer(net['input'], 3, filter_size = (3,3) , pad='valid',W=weight,b = bias))
   it should need weights+bias + beta + gamma+mean+ inv_std, 6 variables. However, when I call 
 lasagne.layers.set_all_param_values(net['conv'],w) (for example, w stores the pre-trained variables ), it says it only need 5 variables, what is missing ? Could anyone give me a favor?

Best Regards
Allen

**Comment by f0k** [comment]
As explained in the [docstring for `batch_norm()`](https://lasagne.readthedocs.io/en/latest/modules/layers/normalization.html#lasagne.layers.batch_norm), when the layer has a bias, `batch_norm()` will remove it since it would be effectless (any bias will be subtracted by the batch normalization).
